packages/validator/src/services/validatorStore.ts

```diff
@@ -252,8 +263,27 @@ export class ValidatorStore {
     delete validatorData["graffiti"];
   }
 
-  getBuilderSelection(pubkeyHex: PubkeyHex): routes.validator.BuilderSelection {
-    return (this.validators.get(pubkeyHex)?.builder || {}).selection ?? this.defaultProposerConfig.builder.selection;
+  getBuilderSelectionParams(pubkeyHex: PubkeyHex): {selection: routes.validator.BuilderSelection; boostFactor: bigint} {
+    const selection =
+      (this.validators.get(pubkeyHex)?.builder || {}).selection ?? this.defaultProposerConfig.builder.selection;
+
+    let boostFactor;
+    switch (selection) {
```

Comment:
Could be a candidate for function extraction for extra clarity

Reply:
this is already a separate fn with single intend to provide builder selection/boost params, lets not increase the call depth